manim/scene/scene.py

```diff
@@ -519,6 +519,47 @@ def remove(self, *mobjects: Mobject):
                 self.restructure_mobjects(mobjects, list_name, False)
             return self
 
+    def replace(self, old_mobject: Mobject, new_mobject: Mobject) -> None:
+        """Replace one Mobject in the scene with another, preserving draw order.
+
+        If old_mobject is a submobject of some other Mobject (e.g. a Group),
+        the new_mobject will replace it inside the group, without otherwise
+        changing the parent mobject.
+
+        Parameters
+        ----------
+        old_mobject - A Mobject which must be in the scene. This method asserts if
+                      old_mobject is not in the scene
+        new_mobject - A Mobject which must not already be in the scene.
+
+        """
+        assert old_mobject is not None
+        assert new_mobject is not None
+
+        def replace_in_list(
+            mobj_list: list[Mobject], old_m: Mobject, new_m: Mobject
+        ) -> bool:
+            for i in range(0, len(mobj_list)):
+                # Is this the old mobject?
+                if mobj_list[i] == old_m:
+                    # If so, write the new object to the same spot and stop looking.
+                    mobj_list[i] = new_m
+                    return True
+                # It is not this object, so recursively look into submobjects
+                if replace_in_list(mobj_list[i].submobjects, old_m, new_m):
```

Comment:
I'm a little concerned about performance with the depth first traversal approach you are doing here to go through submobjects. As `ReplacementTransform` is generally used for "first level" mobjects (i.e, at the first level of the scene graph), I think that a BFS approach is more suited for this problem. I would avoid unnecessarily going through numerous submobjects.
It can be a big game changer for scenes with heavy nested mobjects, such as `StreamLines`, for example.

Reply:
Good point. I picked DFS somewhat arbitrarily, but BFS is just as easy to do.